TD 2659 Update NumberField and FileUploaders

jegasriskantha opened the pull request:
contributes to [TD-2659](https://sce.myjetbrains.com/youtrack/issue/TD-2659/Create-Step-2)

## Changes

 - Updates NumberField to ensure value remains undefined if no value is passed, which also enables default value to be passed into the component. 
 - Updates the ImageUploader and FileUploader to optional accept an id as an input, and updates the onDelete callback to return the deleted file as a second argument. This was done so that when a file uploader accepts multiple files we can accurately determine which file needs to be deleted by it's id.  Where this updated component has been implemented, we are saving the file in a session where the id references a temporary folder where the file is stored. When the user deletes the file, the the deleted file (and it's id) is send to the server so that we can delete only the file that matches that id from the session. 

## Dependencies

N/A

## UI/UX

No real UI updates other than NumberField can now accept a default value. 

## Testing notes

Automated Testing: 
- Added test to NumberField to confirm that default value is rendered when default value is passed in. 


## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [x] Branch has been run in docker.
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

syedsalehinipg commented on src/NumberField/NumberField.stories.jsx:
When I update the default value using the prop, I don't see the value reflected in the field. 

Here I have change the value to 5 but the field is still showing as 12:

![image](https://github.com/IPG-Automotive-UK/react-ui/assets/143519265/bd4c4cfe-f47f-46a6-8351-1d2c06bb0c9b)

jegasriskantha replied:
This default value only works on first render, if you refresh the page you'll see that it has been updated.

syedsalehinipg commented on src/NumberField/NumberField.jsx:
If a `min` and `max` are set, but the default value is not within that range, we should handle this?

Maybe something like this:

```
const [currentValue, setCurrentValue] = useState(() => {
  let initialVal = defaultValue || min;
  if (initialVal < min) initialVal = min;
  if (initialVal > max) initialVal = max;
  return initialVal;
});
```

In this code, the initial value is set to the `defaultValue` or `min` if `defaultValue` is not provided. Then, if the initial value is less than `min`, it's set to `min`. If it's greater than `max`, it's set to `max`. This ensures that the initial value is always within the specified range.

jegasriskantha replied:
HTML standards don't validate on on default value, so I don't think we should either.

syedsalehinipg reviewed (changes requested):
I am unsure how to test the and verify the changes to the FileUploader and ImageUploader,

This is what I attempted:

1. Uploaded two files
2. Deleted one - I see in the actions one object with data and file
3. Deleted next one - the response is a blank array

Should I be seeing a object for each one, to identify which one was deleted?

![image](https://github.com/IPG-Automotive-UK/react-ui/assets/143519265/744ea19f-3ea2-41b2-aa4a-6c1a8f07e10f)

Can I have some more instructions on how to test this please?

jegasriskantha commented on src/NumberField/NumberField.jsx:
No, the current value shouldn't be set at all until the user has interacted. Once the value is set it becomes a controlled component.

jegasriskantha commented:
> I am unsure how to test the and verify the changes to the FileUploader and ImageUploader,
> 
> This is what I attempted:
> 
> 1. Uploaded two files
> 2. Deleted one - I see in the actions one object with data and file
> 3. Deleted next one - the response is a blank array
> 
> Should I be seeing a object for each one, to identify which one was deleted?
> 
> ![image](https://private-user-images.githubusercontent.com/143519265/323616170-744ea19f-3ea2-41b2-aa4a-6c1a8f07e10f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM0NDgyNzMsIm5iZiI6MTcxMzQ0Nzk3MywicGF0aCI6Ii8xNDM1MTkyNjUvMzIzNjE2MTcwLTc0NGVhMTlmLTNlYTItNDFiMi1hYTRhLTZjMWE4ZjA3ZTEwZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxOFQxMzQ2MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNmYzMmZjZGFjMzg4MGYxYmM3MmU1MTZiMGUzZDQ2MWI1OTg2ZjJhNGVlMzI4Mzc2YzJhN2YxM2Y3OGM0YThhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.rwa_nEQcKhq7HTPzUh6O3rGQrlPE8oguPKox_YTJpTU)
> 
> Can I have some more instructions on how to test this please?

Sorry I didn't provide more info. 

I have update both the ImageUploader and FileUploader stories so that the onDelete action returns both arguments. You'll now see that when you delete a file you get both the current selection and the deleted file object.  This is the way to check that the second argument is the deleted object. 

I have also create an new story on FileUploader that passes in the optional id field, you should now see this id when when you delete the file. This is the way to check that the id is passed through the component.

syedsalehinipg reviewed (approved):
LGTM
